### PR TITLE
root: 6.26.06 -> 6.26.08

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Also fixed build on aarch64-darwin.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/misc/root/default.nix [hunk 1]
```diff
@@ -57,11 +57,13 @@ in
 
 stdenv.mkDerivation rec {
   pname = "root";
-  version = "6.26.06";
+  version = "6.26.08";
 
-  src = fetchurl {
-    url = "https://root.cern.ch/download/root_v${version}.source.tar.gz";
-    hash = "sha256-sfc8l2pYClxWyMigFSWCod/FYLTdgOG3VFI3tl5sics=";
+  src = fetchFromGitHub {
```
Comment (veprbl) [inline pkgs/applications/science/misc/root/default.nix:62]: Ant specific reason to use `fetchFromGitHub`(`fetchzip`)? Nice thing about `fetchurl` is that it mirrors to `tarballs.nixos.org`.

**Comment by veprbl** [comment]
cc @ShamrockLee

File: pkgs/applications/science/misc/root/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/applications/science/misc/root/default.nix:62]: This release is only tagged on GitHub. Meanwhile, Fedora and OpenSUSE have all bumped to this version.

File: pkgs/applications/science/misc/root/default.nix [hunk 2]
```diff
@@ -200,6 +204,6 @@ stdenv.mkDerivation rec {
 
     # See https://github.com/NixOS/nixpkgs/pull/192581#issuecomment-1256860426
     # for some context on issues on aarch64.
-    broken = stdenv.isAarch64;
+    broken = stdenv.isAarch64 && stdenv.isLinux;
```
Comment (ShamrockLee) [inline pkgs/applications/science/misc/root/default.nix:207]: Does it really break on Linux? Linux is the primary platform ROOT is used on IMO.

File: pkgs/applications/science/misc/root/default.nix [hunk 3]
```diff
@@ -125,6 +127,8 @@ stdenv.mkDerivation rec {
     # Eliminate impure reference to /System/Library/PrivateFrameworks
     substituteInPlace core/CMakeLists.txt \
       --replace "-F/System/Library/PrivateFrameworks" ""
+  '' + lib.optionalString (lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11" && stdenv.isDarwin) ''
```
Comment (ShamrockLee) [inline pkgs/applications/science/misc/root/default.nix:130]: Though `stdenv.hostPlatform.darwinMinVersion` still evaluates to a string, it sounds more sensible to check `stdenv.isDarwin` first and then Darwin-specific conditions.

```suggestion
  '' + lib.optionalString (stdenv.isDarwin && lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11") ''
```

**Review by ShamrockLee — CHANGES_REQUESTED** [review]
Thanks for the fix! I'll see if the update makes aarch64-linux build.

Update: Just deleted the false suggestions. Sorry for the trouble.

File: pkgs/applications/science/misc/root/default.nix (hunk 3 above)
Comment (wegank) [reply pkgs/applications/science/misc/root/default.nix]: Done.

**Review by ShamrockLee — APPROVED** [review]

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 200412` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>python310Packages.hepmc3</li>
    <li>python39Packages.hepmc3</li>
    <li>root</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Pull request merged**